python/llm/README.md

```diff
@@ -1,140 +1,137 @@
+  input_ids = tokenizer.encode(input_str, ...)
+  output_ids = model.generate(input_ids, ...)
+  output = tokenizer.decode(output_ids, ...)
+  ```
+
+  See the example here.
+
+#### LangChain API
+You may convert Hugging Face *Transformers* models into native INT4 format (currently only *llama*/*bloom*/*gptneox*/*starcoder* model family is supported), and then run the converted models using the LangChain API in `bigdl-llm` as follows.
 
 ```python
-from bigdl.llm.transformers import AutoModelForCausalLM
+from bigdl.llm.langchain.llms import BigdlLLM
+from bigdl.llm.langchain.embeddings import BigdlLLMEmbeddings
+from langchain.chains.question_answering import load_qa_chain
 
-# option 1: load huggingface checkpoint
-llm = AutoModelForCausalLM.from_pretrained("/path/to/llama-7b-hf/",
-                                           model_family="llama")
+embeddings = BigdlLLMEmbeddings(model_path='/path/to/converted/model.bin',
+                                model_family="llama",...)
+bigdl_llm = BigdlLLM(model_path='/path/to/converted/model.bin',
+                     model_family="llama",...)
 
-# option 2: load from huggingface hub repo
-llm = AutoModelForCausalLM.from_pretrained("decapoda-research/llama-7b-hf",
-                                           model_family="llama")
+doc_chain = load_qa_chain(bigdl_llm, ...)
+doc_chain.run(...)
 ```
+See the example here.
 
-You can then use the the model the same way as you use transformers.
-```python
-# Use transformers tokenizer
-tokenizer = AutoTokenizer.from_pretrained(model_ckpt)
-tokens = tokenizer("what is ai").input_ids
-tokens_id = llm.generate(tokens, max_new_tokens=32)
-tokenizer.batch_decode(tokens_id)
-```
+#### `llama-cpp-python`-style API
 
-#### llama-cpp-python like API
-`llama-cpp-python` has become a popular pybinding for `llama.cpp` program. Some users may be familiar with this API so `bigdl-llm` reserve this API and extend it to other model families (e.g., gptneox, bloom)
+You may also run the converted models using the `llama-cpp-python`-style API in `bigdl-llm` as follows.
 
 ```python
 from bigdl.llm.models import Llama, Bloom, Gptneox
 
-llm = Llama("/path/to/llama-7b-int4/bigdl-llm-xxx.bin", n_threads=4)
+llm = Llama("/path/to/converted/model.bin", n_threads=4)
 result = llm("what is ai")
 ```
 
-#### langchain integration
-TODO
-
-## Examples
-We prepared several examples in https://github.com/intel-analytics/BigDL/tree/main/python/llm/example
-
-## Dynamic library BOM
-To avoid difficaulties during the installtion. `bigdl-llm` release the C implementation by dynamic library or executive file. The compilation details are stated below. **These information is only for reference, no compilation procedure is needed for our users.** `GLIBC` version may affect the compatibility.
+### `bigdl-llm` Dependence 
+The native code/lib in `bigdl-llm` has been built using the following tools; in particular, lower  `LIBC` version on your Linux system may be incompatible with `bigdl-llm`.
 
```

Comment:
What is the info for starcoder? @rnwang04

Reply:
info is listed here: https://github.com/intel-analytics/BigDL/pull/8379/files#diff-83b9b2e754ffba10a70a073b16878e1a9f368822ffdb1e4cffc3df70dee84831
Have not merged this PR as action error.